core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -206,41 +191,148 @@ public MutableAggregation apply(StdDev arg) {
   @VisibleForTesting
   static AggregationData createAggregationData(MutableAggregation aggregation) {
     return aggregation.match(
-        new Function<MutableSum, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableSum arg) {
-            return SumData.create(arg.getSum());
-          }
-        },
-        new Function<MutableCount, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableCount arg) {
-            return CountData.create(arg.getCount());
-          }
-        },
-        new Function<MutableHistogram, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableHistogram arg) {
-            return HistogramData.create(arg.getBucketCounts());
-          }
-        },
-        new Function<MutableRange, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableRange arg) {
-            return RangeData.create(arg.getMin(), arg.getMax());
-          }
-        },
-        new Function<MutableMean, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableMean arg) {
-            return MeanData.create(arg.getMean());
-          }
-        },
-        new Function<MutableStdDev, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableStdDev arg) {
-            return StdDevData.create(arg.getStdDev());
-          }
-        });
+        CreateSumData.INSTANCE,
+        CreateCountData.INSTANCE,
+        CreateHistogramData.INSTANCE,
+        CreateRangeData.INSTANCE,
+        CreateMeanData.INSTANCE,
+        CreateStdDevData.INSTANCE);
+  }
+  
+  // static inner Function classes
+  
+  private static final class CreateMutableSum implements Function<Sum, MutableAggregation> {
```

Comment:
Just saw an argument against this from Effective Java, maybe worth considering:

> Do not use wildcard types as return types. Rather than providing additional flexibility for your users, it would force them to use wildcard types in client code.

> Properly used, wildcard types are nearly invisible to users of a class. They cause methods to accept the parameters they should accept and reject those they should reject. If the user of a class has to think about wildcard types, there is probably something wrong with the class’s API.

Also refer to https://stackoverflow.com/questions/34641441/using-generic-wildcard-types-in-return-parameters.

cc @dinooliva